### PR TITLE
fix: fix SmoothScroll not applying custom options

### LINKED ISSUE #11620: SmoothScroll options can't be set
## What should happen?
I should be able to set smoothScroll options via data attributes.
`data-offset="100" data-animation-duration="1"`

## What happens instead?
These attributes are ignored

## Possible Solution
I believe problem is in `_handleLinkClick` method https://github.com/zurb/foundation-sites/blob/develop/js/foundation.smoothScroll.js#L60
`this` is always undefined. So `this.options` is also undefined. So only default options will ever be used.

## Test Case and/or Steps to Reproduce (for bugs)
Codepen in Foundation 6.5.1: https://codepen.io/anon/pen/QJPpbQ
Note that there is no offset and the scroll does take 500ms - instead of 1ms.
Also the scroll stops before actually reaching the element.

## Your Environment

- Foundation version(s) used: 6.5.1

### PR DESCRIPTION
**Pull request by ben-z**
## Description

SmoothScroll no longer takes into account the options passed in to its constructor because the event handler doesn't have the correct `this`. It worked prior to v6.5 because [before the refactor, the event handler uses `_this` from the closure instead of `this` directly](https://github.com/zurb/foundation-sites/blob/140abba37384784f2c6fc38d0ca8b1812c0eba16/js/foundation.smoothScroll.js#L47).

Please refer to https://github.com/zurb/foundation-sites/issues/11620 for a detailed description of this bug.

This PR binds `this` to the event handlers to solve the problem.

- Closes #11620

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (anything that would change an existing functionality)
- [ ] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] The pull request title and template are correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- [x] I have updated the documentation accordingly to my changes (if relevant).
- [ ] I have added tests to cover my changes (if relevant).

It's my first time contributing. As far as I know there aren't any unit tests for this, so I didn't add any tests for my changes. Please let me know if I need to do anything else for this PR!

### CODE REVIEW

**Review by DanielRuf — APPROVED** [review]

**Review by ncoden — CHANGES_REQUESTED** [review]
With `.bind()`, a new function will be created and attached to the event. So when we want to clear the handlers in `_destroy` with `this._handleLinkClick`, the function will not be detached from the event.

You could save `this._handleLinkClick.bind(this)` as `this._linkClickListenner` then reuse it in `_destroy` to clear it.

**Comment by ben-z** [comment]
@ncoden good catch! Fixed as suggested. Since `_events` seems to only be called from the constructor, I don't think we need to worry about `this._linkClickListenner` being overwritten when there are multiple calls to `_events`.

File: js/foundation.smoothScroll.js [hunk 1]
```diff
@@ -39,8 +39,9 @@ class SmoothScroll extends Plugin {
      * @private
      */
     _events() {
-        this.$element.on('click.zf.smoothScroll', this._handleLinkClick.bind(this));
-        this.$element.on('click.zf.smoothScroll', 'a[href^="#"]', this._handleLinkClick.bind(this));
+        this._linkClickListenner = this._handleLinkClick.bind(this);
```
Comment (DanielRuf) [inline js/foundation.smoothScroll.js]: Listener instead of Listenner

**Review by DanielRuf — CHANGES_REQUESTED** [review]

File: js/foundation.smoothScroll.js (hunk 1 above)
Comment (ben-z) [reply js/foundation.smoothScroll.js]: awks 😅. Fixed

**Review by ncoden — APPROVED** [review]
LGTM. Thank you @ben-z 👍

**Comment by ncoden** [comment]
Waiting for @DanielRuf to approve the changes as well.

**Review by DanielRuf — APPROVED** [review]

**Comment by joeworkman** [comment]
This pull request has been mentioned on **Foundation Open Source Community**. There might be relevant details there:

https://foundation.discourse.group/t/foundation-for-sites-v6-6-0-is-here-farout/30/1

**Pull request merged**